# origin: willy (BitKeeper)
# cset: 1.1572 (2.4) key=423026b8MRaI5S3ifsZAsfFXTtbKRg
# URL: http://linux.bkbits.net:8080/linux-2.4/cset@423026b8MRaI5S3ifsZAsfFXTtbKRg
# inclusion: upstream
# descrition: [PATCH] acpi.h needs <linux/init.h>
# revision date: Wed, 23 Mar 2005 18:05:50 +0900
#
# S rset: ChangeSet|1.1571..1.1572
# I rset: include/asm-i386/acpi.h|1.18..1.19
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
# R: Revised  Manually revised by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/03/10 07:51:36-03:00 willy@w.ods.org 
#   [PATCH] acpi.h needs <linux/init.h>
#   
#   the recent acpi merge in 2.4-BK broke compilation of a few external patches
#   (eg: iptable_string), because it declares check_acpi_pci() as void __init,
#   while linux/init.h is not included.
#   
#   Here is a trivial patch which solves the problem.
# 
# include/asm-i386/acpi.h
#   2005/03/06 11:35:37-03:00 willy@w.ods.org +2 -0
#   acpi.h needs <linux/init.h>
# 
#
===== include/asm-i386/acpi.h 1.18 vs 1.19 =====
--- 1.18/include/asm-i386/acpi.h	2005-02-18 23:53:58 +09:00
+++ 1.19/include/asm-i386/acpi.h	2005-03-06 23:35:37 +09:00
@@ -28,6 +28,8 @@
 
 #ifdef __KERNEL__
 
+#include <linux/init.h>
+
 #define COMPILER_DEPENDENT_INT64   long long
 #define COMPILER_DEPENDENT_UINT64  unsigned long long
 
